backend/src/api/routes/editorRoute.ts

```diff
@@ -7,14 +8,18 @@ import {
   getComponentsByAppId,
   updateComponent
 } from '../../services/editorService';
+import permissionMiddleware from '../middlewares/permissionsMiddleware';
 
 const router = Router();
 
 router
-  .get('/:appId', run(req => getComponentsByAppId(req.params.appId)))
-  .get('/:id', run(req => getComponentById(req.params.id)))
-  .post('/:appId', run(req => addComponent(req.params.appId, req.body)))
-  .put('/:id', run(req => updateComponent(req.params.id, req.body)))
-  .delete('/:id', run(req => deleteComponent(req.params.id)));
+  .get('/:appId', permissionMiddleware([Role.ADMIN, Role.DEVELOPER]),
+    run(req => getComponentsByAppId(req.params.appId)))
```

Comment:
we probably use the same routes for app preview, so here you don't need to secure get routes